app/models/component.rb

```diff
@@ -28,6 +32,10 @@ def self.to_service_metadata(event_id, published_at = Time.now)
         service_providers: service_providers.map(&:to_metadata) }.to_json
   end
 
+  def previous_encryption_certificates
+    encryption_certificates.where.not(id: encryption_certificate.id)
```

Comment:
if the component doesn't have any encryption cert (because it's optional) this will fail to evaluate?

Reply:
This would make it more resilient ```encryption_certificates.where.not(id: encryption_certificate&.id)```